x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/engine_analytics_lens_logic.ts

```diff
@@ -0,0 +1,135 @@
+import { kea, MakeLogicType } from 'kea';
+
+import { DataView } from '@kbn/data-views-plugin/public';
+import { FormulaPublicApi, TypedLensByValueInput } from '@kbn/lens-plugin/public';
+
+import { getLensXYLensAttributes, getMetricLensAttributes } from './lens_data';
+
+interface LensArguments {
+  defaultDataView: DataView | null;
+  formula: FormulaPublicApi;
+  isNoResultsCardVisible?: boolean;
+}
+type LensResponse = TypedLensByValueInput['attributes'];
+
+interface EngineAnalyticsLensValues {
+  metricAttributesNoResults: LensResponse;
+  metricAttributesQueries: LensResponse;
+  resultMetricAttributesNoResults: LensResponse | null;
+  resultMetricAttributesQueries: LensResponse | null;
+  resultXYAttributes: LensResponse | null;
+  xyAttributes: LensResponse;
+}
+interface EngineAnalyticsLensActions {
+  getLensAreaAttributes: (args: LensArguments) => LensArguments;
+  getLensMetricAttributes: (args: LensArguments) => LensArguments;
+  setLensAreaAttributes: (resultXYAttributes: EngineAnalyticsLensValues['resultXYAttributes']) => {
+    resultXYAttributes: EngineAnalyticsLensValues['resultXYAttributes'];
+  };
+  setLensMetricAttributesNoResults: (
+    resultMetricAttributesNoResults: EngineAnalyticsLensValues['resultMetricAttributesNoResults']
+  ) => {
+    resultMetricAttributesNoResults: EngineAnalyticsLensValues['resultMetricAttributesNoResults'];
+  };
+  setLensMetricAttributesQueries: (
+    resultMetricAttributesQueries: EngineAnalyticsLensValues['resultMetricAttributesQueries']
+  ) => {
+    resultMetricAttributesQueries: EngineAnalyticsLensValues['resultMetricAttributesQueries'];
+  };
+}
+
+export const EngineAnalyticsLensLogic = kea<
+  MakeLogicType<EngineAnalyticsLensValues, EngineAnalyticsLensActions>
+>({
+  actions: {
+    getLensAreaAttributes: (args: LensArguments) => args,
+    getLensMetricAttributes: (args: LensArguments) => args,
+    setLensAreaAttributes: (resultXYAttributes: LensResponse) => ({
+      resultXYAttributes,
+    }),
+    setLensMetricAttributesNoResults: (resultMetricAttributesNoResults: LensResponse) => ({
+      resultMetricAttributesNoResults,
+    }),
+    setLensMetricAttributesQueries: (resultMetricAttributesQueries: LensResponse) => ({
+      resultMetricAttributesQueries,
+    }),
+  },
+
+  listeners: ({ actions }) => ({
+    getLensAreaAttributes: (input) => {
+      actions.setLensAreaAttributes(
+        getLensXYLensAttributes(
+          input.defaultDataView,
+          input.formula,
+          !input?.isNoResultsCardVisible
+        )
+      );
+    },
+    getLensMetricAttributes: (input) => {
+      actions.setLensMetricAttributesQueries(
+        getMetricLensAttributes(
+          input.defaultDataView,
+          input.formula,
+          !input?.isNoResultsCardVisible
+        )
+      );
+      actions.setLensMetricAttributesNoResults(
+        getMetricLensAttributes(
+          input.defaultDataView,
+          input.formula,
+          input?.isNoResultsCardVisible!
```

Comment:
question: why the ! here? Also, input shouldn't need a ? here because it's definitely defined--otherwise the above two lines would error out.